XenModel/XenAPI-Extensions/Pool.cs

```diff
@@ -95,6 +95,21 @@ public bool IsMasterUpgraded()
             return false;
         }
 
+        public bool IsUnlicensedPool
+        {
+            get
+            {
+
+                bool unlicensed = true;
+                
+                if (Connection != null && Connection.Cache != null && Connection.Cache.Hosts != null
+                    && Connection.Cache.Hosts.All(h => Host.GetEdition(h.edition) != Host.Edition.Free && h.LicenseExpiryUTC() > DateTime.UtcNow - h.Connection.ServerTimeOffset))

```

Comment:
I don't think that Cache and Cache,Hosts can ever be null, so there is no need for these chacks.